### ISSUE TITLE
вопрос по поводу доступа пользователей

### ISSUE THREAD

**Issue by yesss123** [LTD-Beget/sprutio#73]
Как ограничить пользователям доступ на ресурсы. chroot не помогает, jails тоже. в менеджере они могут выходить за пределы домашней директории. и при открытии менеджера открыватся все что есть на сервере, все папки видно. подскажите куда копать?

**Comment by shadowlmd** [comment]
Права правильные ставить нужно, тогда всё будет хорошо. Пользователь работает от своего имени с теми доступами, которые ему предоставляет система. Достаточно сделать `chmod o-rw / /home; chmod o-rwx /home/*`, чтобы пользователь не мог получить листинг файлов в корне и в /home, а так же не мог зайти директорию другого пользователя. При желании пользователь сможет зайти куда-нибудь в /etc, например. Это не страшно, но если вопрос принципиальный, можно и туда, ровно как и в остальные системные директории, доступ закрыть (всё тем же `chmod o-rwx`).

**Comment by yesss123** [comment]
Да, спасибо, опытным путем доперло, что права на папки надо выставлять. Теперь пользователь выше своей папки не ходит. Но в менеджере есть кнопка "корневая директория". Вот при нажатии на нее пользователь выходит на корень сервера и соответственно везде может ходить. Можно эту кнопку отключить что-ли? убрать с формы менеджера? только где она формируется? код то нашли <a class="x-btn x-unselectable x-box-item x-toolbar-item x-btn-default-toolbar-small" hidefocus="on" unselectable="on" id="button-1096" tabindex="0" data-qtip="Корневая директория" componentid="button-1096" style="right: auto; left: 178px; top: 0px; margin: 0px;"><span id="button-1096-btnWrap" data-ref="btnWrap" role="presentation" unselectable="on" style="" class="x-btn-wrap x-btn-wrap-default-toolbar-small "><span id="button-1096-btnEl" data-ref="btnEl" role="presentation" unselectable="on" style="" class="x-btn-button x-btn-button-default-toolbar-small  x-btn-no-text x-btn-icon x-btn-icon-left x-btn-button-center "><span id="button-1096-btnIconEl" data-ref="btnIconEl" role="presentation" unselectable="on" class="x-btn-icon-el x-btn-icon-el-default-toolbar-small fm-action-root " style="">&nbsp;</span><span id="button-1096-btnInnerEl" data-ref="btnInnerEl" unselectable="on" class="x-btn-inner x-btn-inner-default-toolbar-small">&nbsp;</span></span></span></a> , а чем он формируется?

**Comment by shadowlmd** [comment]
Если сделать `chmod o-rw /`, то никто в корень ходить не сможет. Технически, сможет, конечно, но сделать листинг не получится, поэтому Спрут сообщит об ошибке и всё.

**Comment by octopurpie** [comment]
А все таки, как настроить рут для пользователя нормально? а то нерабочая кнопка "Корневая директория" и шанс что кто то телепатически сможет узнать путь и зайти к соседнему пользователю в гости по прямой ссылке как то не тру вей что ли...
Я так понял из исходников, что можно поднимать отдельные rpc на разных серверах с единой точкой входа. Это так? если да, то есть дока или пример, как это использовать?

**Comment by shadowlmd** [comment]
Защититься на 100% от того, что кто-то сможет зайти в чужую директорию, можно только установкой правильных прав на домашние директории. Достаточно ставить права 750 пользователям, и никто ни по какой прямой ссылке в чужую директорию не зайдёт.
